README.md

```diff
@@ -1,28 +1,111 @@
-
-There are several challenges to shipping origin-level isolation, but the clearest blocker is the impact that such isolation would have on the ability to relax the same-origin policy through `document.domain`. Using this mechanism, same-site-but-cross-origin documents can decide they should be able to synchronously script each other at runtime, which makes it impossible to commit them into distinct processes _a priori_. In addition to preventing broad adoption of origin-level process isolation, this feature is bad in itself, as it significantly complicates both the implementation of the web's security boundary, and the expectations we can reasonably set for web developers.
-
-Despite [admonitions against its usage in the HTML spec](https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction), the setter is unfortunately quite pervasive, affecting at least [~0.4% of page views in Chrome](https://chromestatus.com/metrics/feature/timeline/popularity/2544) (down from ~6% thanks to [Facebook's heroic efforts](https://twitter.com/mikewest/status/1136861248186998784)).
+The solution here is to align the developer- and user-visible high-level
+security boundaries with the low-level, system security boundaries.
+That is, process isolation should be aligned with origins.
+
+This would appear to be easily done, since process isolation is invisible
+to the APIs, and therefore browsers are free to change their allocation
+strategies at will. Except for one particular "trick":
+
+Setting the `document.domain` accessor, which allows a page to modify its
+own origin (within a site, e.g. www.example.org to example.org). This allows
+pages or frames hosted on different origins to move themselves into the
+same origin. A example use case that we have observed are pages that host video
+content in an iframe, where the iframe is served from a different origin than
+the main page. By setting document.domain, these might move themselves into
+the same origin, and then the media player and main page can exchange data.
+
+[Setting document.domain has been deprecated]((https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction)
+for along time, but continues being supported by browsers.
+This forces process allocation to be by site,
+not by origin, because a page *might* use `document.domain` to change their
+origin - within the site - later on. Measurements indicate that [setting
+`document.domain` happens on around 0.4% of page views](https://chromestatus.com/metrics/feature/timeline/popularity/2544)  So 99% of pages do not even use this feature. But because they *might* want
```

Comment:
That 0.5% metric shows the page views in which setting `document.domain` enabled access that wouldn't have been allowed if it hadn't been set. Likewise, https://chromestatus.com/metrics/feature/timeline/popularity/2543 shows the pages in which setting `document.domain` blocked access that would otherwise have been allowed.

https://chromestatus.com/metrics/feature/timeline/popularity/2026 is an order of magnitude higher, and shows the page loads upon which the setter was called, regardless of whether it had an impact on a security decision.

Reply:
Done. I'm still a little confused on the metrics, though... :)